src/passes/DeadArgumentElimination.cpp

```diff
@@ -105,7 +104,7 @@ struct DAEScanner
     }
     if (curr->isReturn) {
       info->hasTailCalls = true;
-      info->tailCallees.insert(curr->target);
+      (*infoMap)[curr->target].hasTailCalls = true;
```

Comment:
Should the field `hasTailCalls` be atomic for this to work? (we do that for `hasUnseenCalls`)

Reply:
yes! good call, for some reason i thought i did that, checked my original PR and missed it there as well 👎

Reply:
@kripken Will `fuzz_opt.py` finish on it own or should i leave it run for a certain amount of time?

Reply:
It runs forever. If you see it run for several thousand iterations, that's probably enough here.

Reply:
@kripken Well i was away for a few hours and it ran 15k iterations but it did find something. Not sure if its related to these changes or not 

<details>
<summary>Details</summary>

```

You found a bug! Please report it with

  seed: 16576256188251122925

and the exact version of Binaryen you found it on, plus the exact Python
version (hopefully deterministic random numbers will be identical).

You can run that testcase again with "fuzz_opt.py 16576256188251122925"

The initial wasm file used here is saved as /root/binaryen/out/test/original.wasm

You can reduce the testcase by running this now:

||||
vvvv


/root/binaryen/build-clang/bin/wasm-reduce --nominal /root/binaryen/out/test/original.wasm '--command=bash /root/binaryen/out/test/reduce.sh' -t /root/binaryen/out/test/t.wasm -w /root/binaryen/out/test/w.wasm


^^^^
||||

Make sure to verify by eye that the output says something like this:

At least one of the next two values should be 0:
  0
  1
The following value should be 1:
  1

(If it does not, then one possible issue is that the fuzzer fails to write a
valid binary. If so, you can print the output of the fuzzer's first command
(using -ttf / --translate-to-fuzz) in text form and run the reduction from that,
passing --text to the reducer.)

You can also read "/root/binaryen/out/test/reduce.sh" which has been filled out for you and includes
docs and suggestions.

After reduction, the reduced file will be in /root/binaryen/out/test/w.wasm
================================================================================
```
<details>

Reply:
Hmm, I don't see an error locally with that seed. Can you paste the full output of just running that one seed, `./scripts/fuzz_opt.py 16576256188251122925` ?

Reply:
Ok its passing for me, its possible i had binaries from my follow up PR and maybe there is an issue you there. I will check it that before i post it. 

Just ran the fuzz_opt.py again for 3500 integrations and no issues